src/test/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/resource/GlobalSearchResourceTest.kt

```diff
@@ -0,0 +1,278 @@
+package uk.gov.justice.digital.hmpps.prisonersearch.resource
+
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.springframework.web.reactive.function.BodyInserters
+import uk.gov.justice.digital.hmpps.prisonersearch.QueueIntegrationTest
+import uk.gov.justice.digital.hmpps.prisonersearch.services.Gender
+import uk.gov.justice.digital.hmpps.prisonersearch.services.GlobalSearchCriteria
+import uk.gov.justice.digital.hmpps.prisonersearch.services.SearchCriteria
+import java.time.LocalDate
+
+class GlobalSearchResourceTest : QueueIntegrationTest() {
+
+  companion object {
+    var initialiseSearchData = true
+  }
+
+  @BeforeEach
+  fun setup() {
+
+    if (initialiseSearchData) {
+
+      setupIndexes()
+      indexPrisoners()
+
+      webTestClient.put().uri("/prisoner-index/mark-complete")
+        .headers(setAuthorisation(roles = listOf("ROLE_PRISONER_INDEX")))
+        .exchange()
+        .expectStatus().isOk
+
+      initialiseSearchData = false
+    }
+  }
+
+  @Test
+  fun `access forbidden when no authority`() {
+
+    webTestClient.post().uri("/global-search")
+      .header("Content-Type", "application/json")
+      .exchange()
+      .expectStatus().isUnauthorized
+  }
+
+  @Test
+  fun `access forbidden when no role`() {
+
+    webTestClient.post().uri("/global-search")
+      .body(BodyInserters.fromValue(gson.toJson(SearchCriteria("A7089EY", "john", "smith", "MDI"))))
+      .headers(setAuthorisation())
+      .header("Content-Type", "application/json")
+      .exchange()
+      .expectStatus().isForbidden
+  }
+
+  @Test
+  fun `bad request when no criteria provided`() {
+
+    webTestClient.post().uri("/global-search")
+      .body(BodyInserters.fromValue(gson.toJson(SearchCriteria(null, null, null))))
+      .headers(setAuthorisation(roles = listOf("ROLE_GLOBAL_SEARCH")))
+      .header("Content-Type", "application/json")
+      .exchange()
+      .expectStatus().isBadRequest
+  }
+
+  @Test
+  fun `can perform a match on prisoner number`() {
+    globalSearch(GlobalSearchCriteria("A7089EY", null, null, null, null, null), "/results/globalSearch/search_results_smith.json")
+  }
+
+  @Test
+  fun `can perform a match wrong prisoner number but correct name`() {
+    globalSearch(GlobalSearchCriteria("X7089EY", "JOHN", "SMITH", null, null, null), "/results/globalSearch/search_results_smith.json")
+  }
+
+  @Test
+  fun `can perform a match on PNC number`() {
+    globalSearch(GlobalSearchCriteria("12/394773H", null, null, null, null, null), "/results/globalSearch/search_results_smith.json")
+  }
+
+  @Test
+  fun `can perform a match on CRO number`() {
+    globalSearch(GlobalSearchCriteria("29906/12J", null, null, null, null,null), "/results/globalSearch/search_results_smith.json")
```

Comment:
cool, test is here, might be worth adding another offender with CRO of `29906/99X'  since we had a bug where search was matching any offender on first part of CRO (and PNC) and ignoring the rest